src/app/notes/components/add-note/add-note.component.ts

```diff
@@ -24,32 +24,20 @@ import { Observable, take } from 'rxjs';
 })
 export class AddNoteComponent {
   noteForm: FormGroup;
-  notes$: Observable<Note[]>;
+
   constructor(private fb: FormBuilder, private store: Store) {
     this.noteForm = this.fb.group({
       title: ['', Validators.required],
       description: ['', Validators.required],
     });
-    // Get the current notes from the store
-    this.notes$ = this.store.select(selectNotes);
   }
 
   onSubmit() {
-    if (this.noteForm.valid) {
```

Comment:
I have moved this logic into the reducer itself as we are relying on state to change the outcome. This way, this component doesn't need to know about the current state of notes.